lib/glific/contacts.ex

```diff
@@ -293,11 +293,20 @@ defmodule Glific.Contacts do
   """
   @spec delete_contact(Contact.t()) :: {:ok, Contact.t()} | {:error, Ecto.Changeset.t()}
   def delete_contact(%Contact{} = contact) do
-    if has_permission?(contact.id),
-      do: Repo.delete(contact),
-      else: raise("Permission denied")
+    with true <- has_permission?(contact.id),
+         organization <- Partners.organization(contact.organization_id) do
+      String.equivalent?(organization.contact.phone, contact.phone)
+      |> check_and_delete_contact(contact)
```

Comment:
@AkhileshNegi can you please match the ids here instead of a phone?

Also I think a simple con statement would work here. 

```
cond do
has_permission?(contact.id) == false -> "permission denice" 
is_org_root_contact?(contact) -> "can't delete"
true -> delete the user
```

Then we will be able to use is_org_root_contact? function on other places as well

Reply:
@pankaj-ag made requested changes